src/main/kotlin/domain/usecase/state/CreationStateUseCase.kt

```
@@ -0,0 +1,30 @@
+package com.berlin.domain.usecase.state
+
+
+import com.berlin.domain.helper.IdGeneratorImplementation
+import com.berlin.domain.repository.StateRepository
+import com.berlin.domain.model.State
+
+class CreationStateUseCase(
+    private val stateRepository: StateRepository,
+    private val idGenerator: IdGeneratorImplementation,
+) {
+    fun createNewState(stateName: String, projectId: String): Result<String> {
+        if (validateStateName(stateName)) {
+            val newState = State(
+                id = idGenerator.generateId(stateName),
+                name = stateName,
+                projectId = projectId
+            )
+            return stateRepository.addState(newState)
+                .map { "State created successfully" }
+                .recover { "Creation Failed" }
+        } else {
+            throw Exception("State Name must not be empty or blank")
```

I think it would be better to throw a custom exception here